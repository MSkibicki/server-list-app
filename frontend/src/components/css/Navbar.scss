@import "./Global.scss";

.navbar {
  background-color: $nav-color;
  height: 8vh;
  display: flex;
  align-items: center;
  justify-content: flex-start;

  h1 {
    color: $white;
    font-family: $nav-font-family;
    font-weight: 700;
    margin: 0;
    border-right: 2px $nav-border-color solid;
    width: 11%;
    padding-left: 2.5em;
    font-size: $nav-font;
  }
}
