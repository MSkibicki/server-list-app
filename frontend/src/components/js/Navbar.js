import React from "react";
import "../css/Navbar.scss";

const Navbar = () => {
  return (
    <nav className="navbar">
      <h1>Recruitment task</h1>
    </nav>
  );
};

export default Navbar;
