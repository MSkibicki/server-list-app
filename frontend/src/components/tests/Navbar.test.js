import Navbar from "./../js/Navbar";

describe("<Navbar />", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Navbar />);

    expect(wrapper).toMatchSnapshot();
  });

  it("has correct nav classname", () => {
    const wrapper = shallow(<Navbar />);

    const divClass = wrapper.find("nav").hasClass("navbar");

    expect(divClass).toBe(true);
  });

  it("displays h1 text correctly", () => {
    const wrapper = shallow(<Navbar />);

    const text = wrapper.find("h1").text();

    expect(text).toEqual("Recruitment task");
  });
});
