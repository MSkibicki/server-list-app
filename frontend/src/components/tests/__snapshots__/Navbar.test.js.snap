// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> renders correctly 1`] = `
<nav
  className="navbar"
>
  <h1>
    Recruitment task
  </h1>
</nav>
`;
